src/cascadia/TerminalSettingsModel/FolderEntry.cpp

```diff
@@ -43,6 +49,72 @@ winrt::com_ptr<NewTabMenuEntry> FolderEntry::FromJson(const Json::Value& json)
     JsonUtils::GetValueForKey(json, NameKey, entry->_Name);
     JsonUtils::GetValueForKey(json, IconKey, entry->_Icon);
     JsonUtils::GetValueForKey(json, EntriesKey, entry->_Entries);
+    JsonUtils::GetValueForKey(json, InliningKey, entry->_Inlining);
+    JsonUtils::GetValueForKey(json, AllowEmptyKey, entry->_AllowEmpty);
 
     return entry;
 }
+
+// A FolderEntry should only expose the entries to actually render to WinRT,
+// to keep the logic for collapsing/expanding more centralised.
+using NTMEModel = winrt::Microsoft::Terminal::Settings::Model::NewTabMenuEntry;
+IVector<NTMEModel> FolderEntry::Entries() const
+{
+    // We filter the full list of entries from JSON to just include the
+    // non-empty ones.
+    IVector<NTMEModel> result{ winrt::single_threaded_vector<NTMEModel>() };
+
+    for (const auto& entry : _Entries)
+    {
+        if (entry == nullptr)

```

Comment:
qq: how could we get in this state? An entry is present in the list, but is itself not an entry?